Fix implicit declaration of functions.
Explicitly declare function return types and argument types.
--- configure.orig	2004-03-01 17:06:18.000000000 -0600
+++ configure	2021-11-25 09:05:48.000000000 -0600
@@ -2244,6 +2244,7 @@
 #line 2245 "configure"
 #include "confdefs.h"
 #include <ctype.h>
+#include <stdlib.h>
 #define ISLOWER(c) ('a' <= (c) && (c) <= 'z')
 #define TOUPPER(c) (ISLOWER(c) ? 'A' + ((c) - 'a') : (c))
 #define XOR(e, f) (((e) && !(f)) || (!(e) && (f)))
@@ -2646,6 +2647,7 @@
 cat > conftest.$ac_ext <<EOF
 #line 2648 "configure"
 #include "confdefs.h"
+#include <string.h>
 
 char *blubb() { return (char *) strerror (42); }
 
--- inews/clientlib.c.orig	1996-06-06 14:41:07.000000000 -0500
+++ inews/clientlib.c	2021-11-25 09:11:59.000000000 -0600
@@ -14,6 +14,7 @@
 #include "../config.h"
 #endif
 
+#include <arpa/inet.h>
 #include <stdio.h>
 #ifndef FOR_NN
 #include <sys/types.h>
@@ -51,6 +52,7 @@
 #include <netdnet/dnetdb.h>
 #endif
 
+#include "clientlib.h"
 #include "nntp.h"
 
 FILE	*ser_rd_fp = NULL;
@@ -133,6 +135,7 @@
  *			for reading and writing to server.
  */
 
+int
 server_init(machine)
 char	*machine;
 {
@@ -194,6 +197,7 @@
  *	Errors:		Printed via perror.
  */
 
+int
 get_tcp_socket(machine)
 char	*machine;
 {
@@ -218,7 +222,6 @@
         * fails.
         */
        if( (hp = gethostbyname( machine ) ) == NULL ) {
-               unsigned long inet_addr();
                static struct hostent def;
                static struct in_addr defaddr;
                static char *alist[1];
@@ -344,6 +347,7 @@
  *	Errors:		Printed via nerror.
  */
 
+int
 get_dnet_socket(machine)
 char	*machine;
 {
@@ -427,6 +431,7 @@
  *	Side effects:	None.
  */
 
+int
 handle_server_response(response, server)
 int	response;
 char	*server;
@@ -502,6 +507,7 @@
  *	Side effects:	Talks to server, changes contents of "string".
  */
 
+int
 get_server(string, size)
 char	*string;
 int	size;
@@ -537,7 +543,7 @@
  */
 
 void
-close_server()
+close_server(void)
 {
 	char	ser_line[256];
 
--- inews/clientlib.h.orig	1996-06-06 14:41:07.000000000 -0500
+++ inews/clientlib.h	2021-11-25 09:05:48.000000000 -0600
@@ -4,8 +4,11 @@
  * @(#)clientlib.h	1.1	(Berkeley) 1/9/88
  */
 
-extern	char	*getserverbyfile();
-extern	int	server_init();
-extern	void	put_server();
-extern	int	get_server();
-extern	void	close_server();
+extern	char	*getserverbyfile(char *file);
+extern	int	server_init(char *machine);
+extern	int	get_tcp_socket(char *machine);
+extern	int	get_dnet_socket(char *machine);
+extern	int	handle_server_response(int response, char *server);
+extern	void	put_server(char *string);
+extern	int	get_server(char *string, int size);
+extern	void	close_server(void);
--- inews/inews.c.orig	2004-01-28 20:14:19.000000000 -0600
+++ inews/inews.c	2021-11-25 09:12:43.000000000 -0600
@@ -38,6 +38,7 @@
 #endif
 
 #include "conf.h"
+#include "clientlib.h"
 #include "nntp.h"
 
 
@@ -47,6 +48,12 @@
 
 char	host_name[256];
 
+void append_signature(void);
+void gen_frompath(void);
+int strneql(register char *a, register char *b, int n);
+int valid_header(register char *h);
+
+int
 main(argc, argv)
 int	argc;
 char	*argv[];
@@ -181,7 +188,7 @@
  * The rn-style DOTDIR environmental variable is used if present.
  */
 
-append_signature()
+void append_signature(void)
 {
 	char	line[256], sigfile[256];
 	char	*cp;
@@ -254,7 +261,7 @@
  * a From: line in it.
  */
 
-gen_frompath()
+void gen_frompath(void)
 {
 	char	*full_name;
 	char	*cp;
@@ -330,6 +337,7 @@
  *	Side effects:	None.
  */
 
+int
 strneql(a, b, n)
 register char *a, *b;
 int	n;
